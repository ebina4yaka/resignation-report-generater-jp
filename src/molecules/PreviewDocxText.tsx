import React, { ReactElement, useContext } from 'react'
import { createStyles, makeStyles, Theme } from '@material-ui/core/styles'
import { parametersContext } from '../context/useParameters'
import convertDateStringToJapanFormat from '../libs/convertDateStringToJapanFormat'
import PreviewDocxTitle from '../atoms/PreviewDocxTitle'
import PreviewDocxCompany from '../atoms/PreviewDocxCompany'
import PreviewDocxBody from '../atoms/PreviewDocxBody'
import PreviewDocxBottom from '../atoms/PreviewDocxBottom'

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      display: 'flex',
      flexWrap: 'wrap',
      flexDirection: 'column',
      justifyContent: 'center',
      fontSize: '1.2em',
    },
    body: {
      marginTop: theme.spacing(2),
      marginBottom: theme.spacing(2),
    },
  })
)

export default function PreviewDocxText(): ReactElement {
  const context = useContext(parametersContext)
  const {
    name,
    companyName,
    department,
    representativeDirector,
    dateOfNotification,
    reason,
    dateOfRetirement,
  } = context
  const classes = useStyles()
  return (
    <div className={classes.root}>
      <PreviewDocxTitle />
      <PreviewDocxCompany
        companyName={companyName}
        representativeDirector={representativeDirector}
      />
      <div className={classes.body}>
        <PreviewDocxBody
          reason={reason}
          dateOfRetirement={convertDateStringToJapanFormat(dateOfRetirement)}
        />
      </div>
      <PreviewDocxBottom
        dateOfNotification={convertDateStringToJapanFormat(dateOfNotification)}
        department={department}
        name={name}
      />
    </div>
  )
}
