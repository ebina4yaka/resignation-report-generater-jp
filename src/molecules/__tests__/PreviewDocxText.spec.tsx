import renderer from 'react-test-renderer'
import PreviewDocxText from '../PreviewDocxText'

jest.mock('../../atoms/PreviewDocxBody', () => 'MockedPreviewDocxBody')
jest.mock('../../atoms/PreviewDocxBottom', () => 'MockedPreviewDocxBottom')
jest.mock('../../atoms/PreviewDocxCompany', () => 'MockedPreviewDocxCompany')
jest.mock('../../atoms/PreviewDocxTitle', () => 'MockedPreviewDocxTitle')

test('Snapshot test for PreviewDocxText', () => {
  const tree = renderer.create(<PreviewDocxText />).toJSON()
  expect(tree).toMatchSnapshot()
})
