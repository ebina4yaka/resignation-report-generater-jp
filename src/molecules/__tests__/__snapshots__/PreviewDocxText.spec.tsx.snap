// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test for PreviewDocxText 1`] = `
<div
  className="makeStyles-root-1"
>
  <MockedPreviewDocxTitle />
  <MockedPreviewDocxCompany
    companyName=""
    representativeDirector=""
  />
  <div
    className="makeStyles-body-2"
  >
    <MockedPreviewDocxBody
      dateOfRetirement="Invalid Date"
      reason=""
    />
  </div>
  <MockedPreviewDocxBottom
    dateOfNotification="Invalid Date"
    department=""
    name=""
  />
</div>
`;
